pkg/lib/cockpit.d.ts

```diff
@@ -94,6 +94,49 @@ declare module 'cockpit' {
     function channel(options: ChannelOptions & { binary?: false; }): Channel<string>;
     function channel(options: ChannelOptions & { binary: true; }): Channel<Uint8Array>;
 
+    /* === cockpit.spawn ============================= */
+
+    interface DeferredPromise<T> extends Promise<T> {
+        /* jQuery Promise compatibility */
+        done(callback: (data: T) => void): DeferredPromise<T>
+        fail(callback: (exc: Error) => void): DeferredPromise<T>
+        always(callback: () => void): DeferredPromise<T>
+        progress(callback: (message: T, cancel: () => void) => void): DeferredPromise<T>
+    }
+
+    interface Deferred<T> {
+        resolve(): Deferred<T>;
+        reject(): Deferred<T>;
+        notify(): Deferred<T>;
+        promise: DeferredPromise<T>
+    }
+
+    interface Spawn<T> extends DeferredPromise<T> {
+        input(message: T, stream?: boolean): DeferredPromise<T>;
+        stream(callback: (data: T) => void): DeferredPromise<T>;
+        close(): void;
+    }
+
+    interface SpawnOptions {
+        binary?: boolean,
+        directory?: string;
+        err?: "out" | "ignore" | "message" | "pty";
+        environ?: string[];
+        pty?: boolean;
+        batch?: number;
+        latency?: number
+        superuser?: "try" | "require";
+    }
+
+    function spawn(
+        args: string[],
+        options?: SpawnOptions & { binary?: false }
+    ): Spawn<string>
```

Comment:
Those deserve semicolons...

Why isn't eslint complaining about this?

Reply:
No idea honestly :o